web/scripts/widgets.js

```diff
@@ -11,103 +11,58 @@ function getNumberDefaults(inputData, defaultStep) {
-						targetWidget.value = 0;
-						console.log("increment");
-					} else {
-						targetWidget.value += 1;
-						console.log("increment");
-					}
-				} else {*/
-					if (max) {
-						if (w == "denoise") {
-							targetWidget.value += 0.1;
-							console.log("denoise decrement");
-						} else if (w == "cfg") {
-							targetWidget.value += 0.5;
-							console.log("cfg increment");
-						} else if (w == "width" || w == "height") {
-							targetWidget.value += 64;
-							console.log("width/height decrement");
-						} else {
-							targetWidget.value += 1.0;
-							console.log("decrement");
-						}
-					}
-				}
-				break;
-			case ("decrement"):
-				if (min != null || max != null) {
-					/*if (min) { //Loop to highest and continue batch
-						targetWidget.value = 1125899906842624;
-						console.log("decrement");
-					} else {
-						targetWidget.value -= 1;
-						console.log("decrement");
-					} else {*/
-					if (max) {
-						if (w == "denoise") {
-							targetWidget.value -= 0.1;
-							console.log("denoise decrement");
-						} else if (w == "cfg") {
-							targetWidget.value -= 0.5;
-							console.log("cfg decrement");
-						} else if (w == "width" || w == "height") {
-							targetWidget.value -= 64;
-							console.log("width/height decrement");
-						} else {
-							targetWidget.value -= 1.0;
-							console.log("decrement");
-						}
-
-					}
-				}
-				break;
-			case ("randomize"):
-				if (min != null || max != null) {
-					switch (w) {
-						case ("cfg"):
-						case ("denoise"):
-							console.log("random denoise");
-							targetWidget.value = parseFloat((Math.floor(Math.random() * ((range * 100) + 1)) / 100).toFixed(2));
-							break;
-						case ("height"):
-						case ("width"):
-							targetWidget.value = Math.floor(Math.random() * range) * 64 + min;
-							console.log("Random resolution");
-							break;
-						default:
-							targetWidget.value = Math.floor(Math.random() * range);
-							console.log("Random");
-							break;
-					}
-				}
-				break;
-			default:
-				console.log("default (failed to detect input!)");
-				break;
-		}
-
+    const valueControl = node.addWidget("combo", "control_after_generate", defaultValue, function (v) { }, {
+        values: ["fixed", "increment", "decrement", "randomize"],
+        serialize: false, // Don't include this in prompt.
+    });
+    valueControl.afterQueued = () => {
+
+        var v = valueControl.value;
+        var w = targetWidget.type;
+
+        let min = targetWidget.options?.min;
+        let max = targetWidget.options?.max;
+        let range = Math.max(min, max);
+
+        //adjust values based on valueControl Behaviour
+        switch (v) {
+            case "fixed":
+                break;
+            case "increment":
+                if (min != null && max != null) {
+                    if (max) {
+                        targetWidget.value += targetWidget.options?.increment;
```

Comment:
Try this instead: targetWidget.value += targetWidget.options.step / 10;

Reply:
There is also the issue with picking up width and height for randomized values (haven't found an alternate way of detecting them specifically). Would you rather it be completely random or always divisible by 64?